canfdtest.c

```diff
@@ -123,17 +123,40 @@ static void millisleep(int msecs)
 	if (msecs <= 0) {
 		sched_yield();
 	} else {
-		struct timespec rqtp, rmtp;
-
-		/* sleep in ms */
-		rqtp.tv_sec = msecs / 1000;
-		rqtp.tv_nsec = (msecs % 1000) * 1000000;
-		while (nanosleep(&rqtp, &rmtp)) {
-			if (errno != EINTR) {
-				printf("t\n");
-				break;
+		struct timespec curtp;
+		int err;
+
+		err = clock_gettime(CLOCK_MONOTONIC, &curtp);
```

Comment:
You are right. I missed that.

I have made the changes. I think I messed up the flow by doing a "push -f" on my fork. The pull request shows my latest changes. Is this OK?

Thanks,
Ramesh